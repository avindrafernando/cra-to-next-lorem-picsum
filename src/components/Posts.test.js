import React from "react";
import { render, waitForElement} from "@testing-library/react";
import axiosMock from "axios";
import Posts from "./Posts";

jest.mock("axios");

test("Posts Component", async () => {
  // Arrange
  const returnData = [
    {
      author: "John Doe",
      author_url: "https://unsplash.com/@johndoe",
      filename: "0000_yC-Yzbqy7PY.jpeg",
      format: "jpeg",
      height: 3744,
      id: 0,
      post_url: "https://unsplash.com/photos/yC-Yzbqy7PY",
      width: 5616
    },
    {
      author: "Jane Doe",
      author_url: "https://unsplash.com/@janedoe",
      filename: "0100_pwaaqfoMibI.jpeg",
      format: "jpeg",
      height: 1656,
      id: 1,
      post_url: "https://unsplash.com/photos/pwaaqfoMibI",
      width: 2500
    }
  ];

  axiosMock.get.mockResolvedValueOnce({ data: returnData });
  
  // Act
  // const {getByTitle, container} = render(<Posts/>);

  // const [cardNode1, cardNode2] = await waitForElement(() =>
  //   [getByTitle("John Doe"), getByTitle("Jane Doe")]
  // );

  const {findByTitle, container} = render(<Posts/>);

  const cardNode1 = await findByTitle("John Doe");
  const cardNode2 = await findByTitle("Jane Doe");

  // Assert
  expect(axiosMock.get).toHaveBeenCalledTimes(1);
  expect(axiosMock.get).toHaveBeenCalledWith("https://picsum.photos/list");
  
  expect(cardNode1).toHaveStyle("backgroundImage: 'url(\"https://picsum.photos/250/375?image=0\")'");
  expect(cardNode2).toHaveStyle("backgroundImage: 'url(\"https://picsum.photos/250/375?image=1\")'");

  // snapshots work great with regular DOM nodes!
  expect(container.firstChild).toMatchSnapshot();
});
