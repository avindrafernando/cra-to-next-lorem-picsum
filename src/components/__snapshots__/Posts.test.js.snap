// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Posts Component 1`] = `
<div
  class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-true"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-2"
          style="background-image: url(https://picsum.photos/250/375?image=0);"
          title="John Doe"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            John Doe
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            @johndoe
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              class="makeStyles-hrefLinksNoStyle-3"
              href="https://unsplash.com/@johndoe"
              rel="noopener noreferrer"
              target="_blank"
            >
              See My Work
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              class="makeStyles-hrefLinksNoStyle-3"
              href="https://unsplash.com/photos/yC-Yzbqy7PY"
              rel="noopener noreferrer"
              target="_blank"
            >
              Image Source
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-root MuiGrid-item MuiGrid-grid-lg-true"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation1 MuiCard-root makeStyles-card-1 MuiPaper-rounded"
    >
      <button
        class="MuiButtonBase-root MuiCardActionArea-root"
        tabindex="0"
        type="button"
      >
        <div
          class="MuiCardMedia-root makeStyles-media-2"
          style="background-image: url(https://picsum.photos/250/375?image=1);"
          title="Jane Doe"
        />
        <div
          class="MuiCardContent-root"
        >
          <h2
            class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom"
          >
            Jane Doe
          </h2>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiTypography-colorTextSecondary"
          >
            @janedoe
          </p>
        </div>
        <span
          class="MuiCardActionArea-focusHighlight"
        />
        <span
          class="MuiTouchRipple-root"
        />
      </button>
      <div
        class="MuiCardActions-root MuiCardActions-spacing"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              class="makeStyles-hrefLinksNoStyle-3"
              href="https://unsplash.com/@janedoe"
              rel="noopener noreferrer"
              target="_blank"
            >
              See My Work
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeSmall"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
            <a
              class="makeStyles-hrefLinksNoStyle-3"
              href="https://unsplash.com/photos/pwaaqfoMibI"
              rel="noopener noreferrer"
              target="_blank"
            >
              Image Source
            </a>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
